herddb-core/src/main/java/herddb/client/ClientConfiguration.java

```diff
@@ -211,21 +211,6 @@ public void readJdbcUrl(String url) {
             set(PROPERTY_MAX_CONNECTIONS_PER_SERVER, 1);
             set(PROPERTY_CLIENT_CONNECT_REMOTE_SERVER, false);
         }
-        if (questionMark < url.length()) {
```

Comment:
If we drop this line users of ClientConfiguration without the JDBC Datasource will see that the querystring is no more processed.

Probably we should update the "properties" field of BasicHerdDBDataSource after calling this method (not that smart but it can work)

https://github.com/diennea/herddb/blob/7148a254efafdc80675ccda547fa15f2507cf586/herddb-jdbc/src/main/java/herddb/jdbc/BasicHerdDBDataSource.java#L214

Reply:
we cant update it after since it must be done before anything else (literal in connect) since it is used by the whole stack
we can always duplicate the parsing or forward client.* properties maybe?
in clientconfig we can have a flag "intiialized", if false -> parse, else noop and initialized would be set to true by setClientProps or setUrl? sthg like that would work better pby

Reply:
works for me.
We can create an utility method in order to not duplicate the code, but not a big deal